src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx

```diff
@@ -52,60 +48,44 @@ t8_geometry_squared_disk::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreei
     return;
   }
 
-  /* Four squares framing the central one. */
-  {
-    const double center_ref[3] = { 0.5, 0.5, 0.0 };
-    t8_geom_linear_interpolation (center_ref, active_tree_vertices, 3, 2, n);
-
-    /* Normalize vector `n`. */
-    const double norm = sqrt (n[0] * n[0] + n[1] * n[1]);
-    n[0] = n[0] / norm;
-    n[1] = n[1] / norm;
-  }
+  /* Normal vector along one of the straight edges of the quad. */
+  n[0] = active_tree_vertices[0 + 0];
+  n[1] = active_tree_vertices[0 + 1];
+  n[2] = active_tree_vertices[0 + 2];
+  t8_vec_normalize (n);
 
-  {
-    /* Radial vector parallel to one of the tilted edges of the quad. */
-    r[0] = active_tree_vertices[0];
-    r[1] = active_tree_vertices[1];
-
-    /* Normalize vector `r`. */
-    const double norm = sqrt (r[0] * r[0] + r[1] * r[1]);
-    r[0] = r[0] / norm;
-    r[1] = r[1] / norm;
-  }
+  /* Radial vector parallel to one of the tilted edges of the quad. */
+  r[0] = active_tree_vertices[9 + 0];
+  r[1] = active_tree_vertices[9 + 1];
+  r[2] = active_tree_vertices[9 + 2];
+  t8_vec_normalize (r);
 
   for (size_t i_coord = 0; i_coord < num_coords; i_coord++) {
     size_t offset = 3 * i_coord;
 
-    const double x = ref_coords[offset + 0];
-    const double y = ref_coords[offset + 1];
+    const double x_ref = ref_coords[offset + 0];
+    const double y_ref = ref_coords[offset + 1];
 
     {
       double corr_ref_coords[3];
 
       /* Correction in order to rectify elements near the corners. */
-      corr_ref_coords[0] = tan (0.5 * M_PI * (x - 0.5)) * 0.5 + 0.5;
-      corr_ref_coords[1] = y;
+      corr_ref_coords[0] = tan (0.25 * M_PI * x_ref);
+      corr_ref_coords[1] = y_ref;
```

Comment:
Do you have a reference, as to why you use this kind of correction?

Reply:
No, I don't. Divine insight resp. by tinkering. I suspect there actually no analytical function which does a perfect correction.